subprojects/docs/src/docs/userguide/reference/ci-systems/github-actions.adoc

```diff
@@ -0,0 +1,149 @@
+
+== What you'll need
+
+* A text editor
+* A command prompt
+* The Java Development Kit (JDK), version 1.8 or higher
+* A https://gradle.org/install[Gradle distribution], version {gradle-version} or better
+* A GitHub account
+
+== Setup a typical project
+
+As example, this guide is going to focus on a Java-based project.
+More specifically, a Gradle plugin written in Java and tested with https://www.spekframework.org/[Spek].
+First, we'll get the project set up on your local machine before covering the same steps on CI.
+
+Just follow these steps:
+
+=== Fork the https://github.com/gradle/gradle-site-plugin[Gradle Site Plugin] repository
+
+Open https://github.com/gradle/gradle-site-plugin and click the "Fork" button top right. Create the fork under your personal GitHub account.
+
+=== Clone the repository locally
+
+[listing.terminal.sample-command]
+----
+$ git clone https://github.com/<github-user>/gradle-site-plugin.git
+Cloning into 'gradle-site-plugin'...
+$ cd gradle-site-plugin
+----
+
+=== Build the project
+
+As a developer of a Java project, you'll typical want to compile the source code, run the tests and assemble the JAR artifact. That's no different for Gradle plugins. The following command achieves exactly that:
+
+[listing.terminal.sample-command]
+----
+$ ./gradlew build
+
+BUILD SUCCESSFUL
+14 actionable tasks: 14 executed
+----
+
+The project provides the <<gradle_wrapper.adoc#gradle_wrapper,Gradle Wrapper>> as part of the repository.
+It is a recommended practice for any Gradle project as it enables your project to built on CI without having to install the Gradle runtime.
+
+=== Build scan integration
+
+The sample project is equipped with support for generating link:https://scans.gradle.com/[build scans].
+Running the build with the command line option `--scan` renders a link in the console.
+
+[listing.terminal.sample-command]
+----
+$ ./gradlew build --scan
+Publishing build scan...
+https://gradle.com/s/7mtynxxmesdio
+----
+
+The following section will describe how to build the project with the help of GitHub Actions.
+
+== Configure GitHub Actions
+
+GitHub Actions is a free, cloud-based CI solution provider built directly into GitHub making it an excellent choice for open source projects hosted on GitHub.
+You can build any project as long as it is hosted on GitHub as public repository.
+GitHub Actions can be configured to make your build outputs and test results available for download.
+
+=== Configure the GitHub Actions workflow
+
+You create a GitHub Actions workflow by adding a `.github/workflows/<workflow-name>.yml` file to your repository.
+This workflow definition file contains all relevant instructions for building the project on GitHub Actions.
+
+The following workflow file instructs GitHub Actions to build your Gradle project with the default JDK, using th Java project with JDK 8, using the Gradle Wrapper to run the build.
```

Comment:
:exclamation: - The double "using" throws me off, I wouldn't repeat it.

```suggestion
The following workflow file instructs GitHub Actions to build your Gradle project using the Gradle Wrapper and executed by the default Java distribution -- currently JDK 1.8.
```

Reply:
:grey_question: - Shouldn't we rather promote JDK 11 instead of JDK 8, as it's EOL?
The action's repository also has 11 as the example version.

Reply:
:x: - at minimum, this typo should be fixed
```suggestion
The following workflow file instructs GitHub Actions to build your Gradle project with the default JDK, using the Java project with JDK 8, using the Gradle Wrapper to run the build.
```

Reply:
I improved the wording and removed any reference to the exact JDK used (it's not important to the tutorial).